ml_genn/model.py

```diff
@@ -408,175 +375,319 @@ def traverse_tf_out_layers(out_layers):
+                    config.shape = tf_layer.output_shape[1:]
+                    config.has_activation = True
+                    config.neurons = converter.create_neurons(tf_layer, pre_compile_output)
+
+                    converter.validate_tf_layer(tf_layer, config)
+
+                    # configure synapses
+                    for in_config in in_configs:
+
+                        if in_config.has_activation:
+                            # configure Identity synapses
+                            config.synapses_class.append(IdentitySynapses)
+                            config.synapses_params.append({})
+                            config.synapses_source.append(in_config)
+                            config.synapses_weights.append(None)
+
+                        else:
+                            for i in range(len(in_config.synapses_class)):
+                                # copy incoming synapses
+                                config.synapses_class.append(in_config.synapses_class[i])
+                                config.synapses_params.append(in_config.synapses_params[i])
+                                config.synapses_source.append(in_config.synapses_source[i])
+                                config.synapses_weights.append(in_config.synapses_weights[i])
+
+                    config_steps.append(config)
+
+                    configs_lookups[tf_layer] = [config]
+
 
                 # === Ignored Layers ===
-                elif isinstance(tf_layer, ignored_tf_layers):
-                    pass
+                elif isinstance(tf_layer, tf_ignored_layers):
+
+                    assert(len(tf_in_layers) == 1)
+                    tf_in_layer = tf_in_layers[0]
+                    in_configs = configs_lookups[tf_in_layer]
+
+                    configs_lookups[tf_layer] = in_configs
+
+
+                # === Unsupported Layers ===
+                else:
+                    raise NotImplementedError('{} layers not supported'.format(
+                        tf_layer.__class__.__name__))
+
+
+        # execute model build process
+        mlg_layer_lookup = {}
+        mlg_model_inputs = []
+        mlg_model_outputs = []
+
+        # for each build step
+        for config in config_steps:
+
+            if config.is_input:
+                # build layer
+                mlg_layer = InputLayer(config.name, config.shape, config.neurons)
+
+                mlg_model_inputs.append(mlg_layer)
+
+            else:
+                # build layer
+                mlg_layer = Layer(config.name, config.neurons)
+
+                # build synapses
+                sources = []
+                synapses = []
+                weights = config.synapses_weights
+                for i in range(len(config.synapses_class)):
+                    sources.append(mlg_layer_lookup[config.synapses_source[i]])
+                    synapses.append(config.synapses_class[i](**config.synapses_params[i]))
```

Comment:
with my suggested change, this would also be reduced to:
```python
sources = [mlg_layer_lookup[s.source]) for s in config.synapses]
synapses= [s.class(**s.params) for s in config.synapses]
```
which is much more pythonic